xcube/core/store/storepool.py

```diff
@@ -0,0 +1,273 @@
+# The MIT License (MIT)
+# Copyright (c) 2020 by the xcube development team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import json
+from typing import Any, Dict, Optional, List
+
+from xcube.util.assertions import assert_given
+from xcube.util.assertions import assert_instance
+from xcube.util.jsonschema import JsonObjectSchema
+from xcube.util.jsonschema import JsonStringSchema
+from .store import DataStore
+from .store import DataStoreError
+from .store import new_data_store
+
+
+def get_data_store(store_id: str,
+                   store_params: Dict[str, Any] = None,
+                   store_pool: 'DataStorePool' = None) -> DataStore:
+    """
```

Comment:
> First, store_id can have two different meanings here

No, because instance arguments are called `store_instance_id`.

> Also, right now the method does not make a lot of sense. 

This method has been designed in response to what is needed (in "xcube gen2") in EDC and in fact - it is already in use. Please have a look at its usages.

> Why not just call "store_pool.get_store(store_instance_id)" if you have both at hand anyway?

Again, please refer to the use case.
You can always use `store_pool.get_store(store_instance_id)` iff this is sufficient in your case.